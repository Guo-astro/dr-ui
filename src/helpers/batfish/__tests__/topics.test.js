const { buildTopics } = require('../topics.js');
const data = require('./fixtures/data.json');

describe('buildTopics', () => {
  it('single structure', () => {
    expect(buildTopics(data)).toMatchSnapshot();
  });
});
