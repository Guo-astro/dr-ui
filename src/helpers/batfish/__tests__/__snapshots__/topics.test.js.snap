// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildTopics single structure 1`] = `
Object {
  "/docs-starter-kit/examples/": Object {
    "contentType": "example",
    "description": "Replace me.",
    "language": Array [
      "JavaScript",
    ],
    "layout": "example",
    "navOrder": 3,
    "path": "/docs-starter-kit/examples/",
    "title": "Examples",
    "topics": Array [
      Object {
        "count": 3,
        "name": "Cool",
        "pages": Array [
          Object {
            "contentType": "example",
            "description": "Replace me.",
            "language": Array [
              "JavaScript",
            ],
            "layout": "example",
            "path": "/docs-starter-kit/examples/example-1/",
            "text": "Example 1",
            "thumbnail": "header-image",
            "title": "Example 1",
            "topic": Array [
              "Cool",
            ],
            "url": "/docs-starter-kit/examples/example-1/",
          },
          Object {
            "contentType": "example",
            "description": "Replace me.",
            "language": Array [
              "JavaScript",
            ],
            "layout": "example",
            "path": "/docs-starter-kit/examples/example-2/",
            "text": "Example 2",
            "thumbnail": "header-image",
            "title": "Example 2",
            "topic": Array [
              "Cool",
            ],
            "url": "/docs-starter-kit/examples/example-2/",
          },
          Object {
            "contentType": "example",
            "description": "Replace me.",
            "language": Array [
              "JavaScript",
            ],
            "layout": "example",
            "path": "/docs-starter-kit/examples/example-3/",
            "text": "Example 3",
            "thumbnail": "header-image",
            "title": "Example 3",
            "topic": Array [
              "Cool",
              "Awesome",
            ],
            "url": "/docs-starter-kit/examples/example-3/",
          },
        ],
        "url": "/docs-starter-kit/examples/#cool",
      },
      Object {
        "count": 1,
        "name": "Awesome",
        "pages": Array [
          Object {
            "contentType": "example",
            "description": "Replace me.",
            "language": Array [
              "JavaScript",
            ],
            "layout": "example",
            "path": "/docs-starter-kit/examples/example-3/",
            "text": "Example 3",
            "thumbnail": "header-image",
            "title": "Example 3",
            "topic": Array [
              "Cool",
              "Awesome",
            ],
            "url": "/docs-starter-kit/examples/example-3/",
          },
        ],
        "url": "/docs-starter-kit/examples/#awesome",
      },
    ],
  },
}
`;
